lib/src/client.dart

```diff
@@ -51,42 +53,154 @@ class ChannelOptions {
   }
 }
 
-/// A channel to an RPC endpoint.
+/// A connection to a single RPC endpoint.
+///
+/// RPCs made on a connection are always sent to the same endpoint.
+class ClientConnection {
+  static final _methodPost = new Header.ascii(':method', 'POST');
+  static final _schemeHttp = new Header.ascii(':scheme', 'http');
+  static final _schemeHttps = new Header.ascii(':scheme', 'https');
+  static final _contentTypeGrpc =
+      new Header.ascii('content-type', 'application/grpc');
+  static final _teTrailers = new Header.ascii('te', 'trailers');
+  static final _grpcAcceptEncoding =
+      new Header.ascii('grpc-accept-encoding', 'identity');
+  static final _userAgent = new Header.ascii('user-agent', 'dart-grpc/0.2.0');
+
+  final ClientTransportConnection _transport;
+
+  bool _isShutdown = false;
+
+  ClientConnection(this._transport);
+
+  static List<Header> createCallHeaders(
+      bool useTls, String authority, String path, CallOptions options) {
+    final headers = [
+      _methodPost,
+      useTls ? _schemeHttps : _schemeHttp,
+      new Header.ascii(':path', path),
+      new Header.ascii(':authority', authority),
+    ];
+    if (options.timeout != null) {
+      headers.add(
+          new Header.ascii('grpc-timeout', toTimeoutString(options.timeout)));
+    }
+    headers.addAll([
+      _contentTypeGrpc,
+      _teTrailers,
+      _grpcAcceptEncoding,
+      _userAgent,
+    ]);
+    options.metadata.forEach((key, value) {
+      headers.add(new Header.ascii(key, value));
+    });
+    return headers;
+  }
+
+  /// Shut down this connection.
```

Comment:
[Effective dart](https://www.dartlang.org/guides/language/effective-dart/documentation#prefer-starting-function-or-method-comments-with-third-person-verbs) would have you say "Shuts down this connection".

Applies several times below.